ipaserver/install/ipa_migrate.py

```diff
@@ -0,0 +1,2028 @@
+            issuer = self.get_cert_issuer(cert_val)
+            REALM_LIST = [self.realm, self.remote_realm]
+            cert_removed = False
+            for realm in REALM_LIST:
+                if issuer == f"CN=Certificate Authority,O={realm}":
+                    # This is an IPA issued cert, remove it
+                    remove_vals.append(cert_val)
+                    self.log_debug("Removed IPA issued userCertificate "
+                                   f"from: {entry_dn}")
+                    cert_removed = True
+                    break
+            if not cert_removed:
+                self.log_debug("Keeping userCertificate issued by "
+                               f"'{issuer}' in entry: {entry_dn}")
+
+        # Now remove the values from cert_vals
+        for val in remove_vals:
+            cert_values.remove(val)
+
+        return len(cert_values) == 0
+
+    def convert_value(self, val, dns=False):
+        """
+        Replace suffix, hostname, domain, and realm from a string
+        """
+        if isinstance(val, bytes) or isinstance(val, DN):
+            return val
+
+        # Replace base DN
+        val = self.replace_suffix_value(val)
+
+        # For DNS DN we only replace suffix
+        if dns:
+            return val
+
+        # Replace host
+        if self.args.hostname in val:
+            val = val.replace(self.args.hostname, self.local_hostname)
+
+        # Replace domain
+        if self.remote_domain in val and self.local_domain not in val:
+            val = val.replace(self.remote_domain, self.local_domain)
+
+        # Replace realm
+        val = val.replace(self.remote_realm, self.realm)
+
+        return val
+
+    def convert_values(self, values, dns=False):
+        """
+        Replace suffix, hostname, domain, and realm in a list
+        """
+        new_values = []
+        for val in values:
+            new_values.append(self.convert_value(val, dns))
+
+        # normalize DN values
+        return self.normalize_vals(new_values)
+
+    def get_ldapentry_attr_vals(self, entry, attr):
+        """
+        Get the raw attribute values from IPA's LDAPEntry
+        """
+        vals = []
+        attr_vals = entry.raw.get(attr)
+        for val in attr_vals:
+            if isinstance(val, bytes):
+                vals.append(ensure_str(val))
+            elif not isinstance(val, str):
+                val = str(val)
+                vals.append(ensure_str(val))
+            else:
+                # Just a string
+                vals.append(val)
+
+        return self.normalize_vals(vals)
+
+    def build_ldap_entry(self, dn, attrs):
+        """
+        Take a DN and some attributes and build an LDAPEntry.  Used when
+        adding entries to the local server
+        """
+        entry = LDAPEntry(self.local_conn, DN(dn))
+        range_reset = False
+        for attr, values in attrs.items():
+            if (self.args.reset_range or self.mode == "stage-mode") and \
```

Comment:
If we're changing UID/GID then the user will need a new SID as well.

Reply:
> If we're changing UID/GID then the user will need a new SID as well.

Would there be a reason to reset the DNA attributes if we are in production mode?  I added that functionality, but not because I had a valid reason.  I was just trying to be versatile, but if I can remove it then this would fix itself.